app/domain/authentication/authn_azure/application_identity.rb

```diff
@@ -0,0 +1,132 @@
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: ConstraintNotSupported, MissingAnnotationConstraint, IllegalConstraintCombinations
+
+    # This class defines an application identity of a given Conjur host.
+    # The constructor initializes an ApplicationIdentity object and validates that
+    # it is configured correctly.
+    # The difference between the validation in this constructor and
+    # the validation in ValidateApplicationIdentity is that here we validate that
+    # the application identity is configured correctly, and thus is a valid application
+    # identity. In ValidateApplicationIdentity we validate that the defined application
+    # identity is actually the one we expect from Azure. This validation is done through
+    # a combination of comparisons that are performed based on the type of assigned
+    # identity provided.
+    # For example, an application identity `some-namespace/blah/some-value` is not
```

Comment:
some-namespace? :)

Please give an example with authn-azure annotations

Reply:
Good catch 🥇 
Adding the following examples for application identities
Not valid - `authn-azure/some-value`
Valid - `authn-azure/subscription-id` or `authn-azure/subscription-id` or `authn-azure/<service-id>/subscription-id`

Reply:
please note that the example `some-namespace/blah/some-value` if for an application identity that is in the id. I suggest writing down a whole host definition with annotations.